File: day-0/python/singleton.py
class Singleton:
    __instance = None

    @staticmethod
    def getInstance():
        # Static access method.
        if Singleton.__instance == None:
            Singleton()
        return Singleton.__instance

    def __init__(self):
        if Singleton.__instance != None:
            raise Exception("This class is a Singleton!")
        else:
            Singleton.__instance = self


s = Singleton()
print(s)

# This will raise an Error
# s2 = Singleton()
# print(s2)

s2 = Singleton.getInstance()
print(s2)

File: day-0/php/Singleton.php
<?php


class Singleton
{
    private static $instance;

    private function __construct()
    {
    }
    private function __clone()
    {
    }
    private function __wakeup()
    {
    }

    public static function getInstance()
    {
        if (null === self::$instance) {
            self::$instance = new self();
            echo "Created the Singleton Object for the first time. \n";
        } else {
            echo "Returning the already created Singleton object. \n";
        }
        return self::$instance;
    }
}

// This will trigger an error
// $s1 = new Singleton();

$s1 = Singleton::getInstance();
$s2 = Singleton::getInstance();
